fe/fe-core/src/main/java/com/starrocks/catalog/PrimitiveType.java

```diff
@@ -68,6 +68,8 @@ public enum PrimitiveType {
 
     FUNCTION("FUNCTION", 8, TPrimitiveType.FUNCTION),
 
+    SUBQUERY("SUBQUERY", -1, TPrimitiveType.SUBQUERY),
```

Comment:
subquery could be expr, but why subquery could be type?

Reply:
We first transform `Expr` to `ScalarOperator`, subquery will be wrapped inside a `SubqueryOperator`, which needs a ScalarType.

BTW, I removed the new primitiveType, and added an flag in scalarType to distinguish